.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,47 @@
+name: CodeQL test
+
+on:
+  pull_request:
+    paths-ignore:
+      - '**/README.md'
+  push:
+    branches:
+      - 'release/**'
+      - 'hotfix/**'
+      - 'alpha-release/**'
+
+jobs:
+  analyse:
+    runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: -Dmaven.wagon.http.retryHandler.count=3 -Dmaven.wagon.httpconnectionManager.ttlSeconds=25
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: '8'
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository
+            !~/.m2/repository/uk
+          key: all-gaffer-dependencies
+          restore-keys: Accumulo-gaffer-dependencies
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: java
+
+      - name: Build Code
+        run: mvn -B -ntp clean install -Pquick -Dskip.jar-with-dependencies=true -Dshaded.jar.phase=true
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v2
```

Comment:
Missing newline

Reply:
This is consistent with all of the other workflows, none of which have newlines at the end of the file.

Reply:
But they should!

Reply:
If this is the case shouldn't it be enforced through linting/stylechecking?

Reply:
We currently don't have linting on the ci/yaml files but probably should. I have read a cool blog about ensuring GitHub Actions are secure with semgrep: https://semgrep.dev/blog/2021/protect-your-github-actions-with-semgrep